### DISCUSSION TITLE
Property with multiple data types

### DISCUSSION THREAD

**Discussion by vishnuamritanshuis** [JanusGraph/janusgraph#3116] [Q&A]
I am using  Janusgraph 0.6.1 in one of application. 
I am facing an issue, while adding adding vertex with same properties key but with diff datatypes.
I have a requirement where we need to provide the flexibility to let users add different datatypes with same property keys.
 

Please check below details where I have replicated the same issue from gremlin console client.

gremlin> JanusGraph.version()
==>0.6.1
gremlin> graph = JanusGraphFactory.open('inmemory')
==>standardjanusgraph[inmemory:[127.0.0.1]]
gremlin> g = graph.traversal()
==>graphtraversalsource[standardjanusgraph[inmemory:[127.0.0.1]], standard]
gremlin> g.addV("Vabc").property("Vprop", 9).iterate()
gremlin> g.tx().commit()
==>null
gremlin> g.addV("Sabc").property("Vprop", "this is a string").iterate()
Value [this is a string value] is not an instance of the expected data type for property key [Vprop] and cannot be converted. Expected: class java.lang.Integer, found: class java.lang.String.

Is there any specific configuration/settings at Janus graph  to fix the same or any other workaround?

Thanks,
Vishnu Amritanshu

**Comment by li-boxuan** [comment]
You can use `Object.class` for a property. See https://docs.janusgraph.org/schema/#property-key-data-type
```
gremlin> graph = JanusGraphFactory.open("inmemory")
==>standardjanusgraph[inmemory:[127.0.0.1]]
gremlin> mgmt = graph.openManagement()
==>org.janusgraph.graphdb.database.management.ManagementSystem@80b122b
gremlin> mgmt.makePropertyKey("Vprop").dataType(Object.class).make()
==>Vprop
gremlin> mgmt.commit()
==>null
gremlin> g = graph.traversal()
==>graphtraversalsource[standardjanusgraph[inmemory:[127.0.0.1]], standard]
gremlin> g.addV("Vabc").property("Vprop", 9).iterate()
gremlin> g.tx().commit()
==>null
gremlin> g.addV("Sabc").property("Vprop", "this is a string").iterate()
gremlin> g.V().valueMap()
09:14:40 WARN  org.janusgraph.graphdb.transaction.StandardJanusGraphTx  - Query requires iterating over all vertices [()]. For better performance, use indexes
==>[Vprop:[this is a string]]
==>[Vprop:[9]]
gremlin>
```
